src/erc721_utils.rs

```diff
@@ -0,0 +1,255 @@
+//! This module contains utility functions for interacting with ERC20 tokens and contracts
+use crate::jsonrpc::error::Web3Error;
+use crate::{client::Web3, types::SendTxOption};
+use clarity::{abi::encode_call, PrivateKey as EthPrivateKey};
+use clarity::{Address, Uint256};
+use num::Bounded;
+use std::time::Duration;
+use tokio::time::timeout as future_timeout;
+
+pub static ERC20_GAS_LIMIT: u128 = 100_000;
+
+impl Web3 {
+    /// Checks if any given contract is approved to spend money from any given erc20 contract
```

Comment:
@jkilpatr 
Thanks for code comments, but I changed this to WIP because I still need to run this in conjunction with main codebase in order to do the transfer and approve stuff.